server/aesdsocket.c

```diff
@@ -0,0 +1,390 @@
+// Author: Sundar Krishnakumar
+// file: aesdsocket.c
+// brief: stream socket server
+// Reference/Re-use: https://beej.us/guide/bgnet/html/
+//            https://www.educative.io/edpresso/how-to-implement-tcp-sockets-in-c
+//            https://codingfreak.blogspot.com/2009/10/signals-in-linux-blocking-signals.html
+//            https://linuxprograms.wordpress.com/2012/06/22/c-getopt_long-example-accessing-command-line-arguments/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <syslog.h>
+#include <getopt.h>
+#include <stdbool.h>
+
+#define PORT 9000
+#define MAX_CONNECTS 5
+#define FILE_NAME "/var/tmp/aesdsocketdata"
+#define BUFFER_SIZE 100
+#define LINE_SZ_TABLE_MAX 10000
+
+
+// global variables
+struct sockaddr_in server_addr, client_addr;
+socklen_t addrlen;
+int server_fd, client_fd, fd;
+sigset_t intmask;
+char buffer[BUFFER_SIZE];
+
+long lin_sz_table[LINE_SZ_TABLE_MAX];
+long line_sz_index = 0;
+bool is_daemon = false;
+
+struct option long_options[] = {
+    {"daemon", no_argument, 0, 'a'},
+    {0, 0, 0, 0}
+};
+
+// get sockaddr, IPv4 or IPv6
+// Code source: https://beej.us/guide/bgnet/html/
+void *get_in_addr(struct sockaddr *sa)
+{
+    if (sa->sa_family == AF_INET) {
+        return &(((struct sockaddr_in*)sa)->sin_addr);
+    }
+
+    return &(((struct sockaddr_in6*)sa)->sin6_addr);
+}
+
+// graceful exit
+void exit_handler(int signum)
+{
+
+    syslog(LOG_DEBUG, "Caught signal, exiting");
+
+    close(fd);	
+	close(server_fd);
+    close(client_fd);
+    closelog();
+
+    // Delete file
+    if(remove(FILE_NAME) < 0) 
```

Comment:
@SundarKrishnakumar are all of these functions safe to call from signal handler?